atom/browser/api/views/atom_api_resize_area.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2018 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/api/views/atom_api_resize_area.h"
+
+#include "atom/common/api/constructor.h"
+#include "native_mate/dictionary.h"
+
+#include "atom/common/node_includes.h"
+
+namespace atom {
+
+namespace api {
+
+ResizeArea::ResizeArea() : View(new views::ResizeArea(this)) {
+  view()->set_owned_by_client();
+}
+
+ResizeArea::~ResizeArea() {}
+
+void ResizeArea::OnResize(int resize_amount, bool done_resizing) {
+  Emit("resize", resize_amount, done_resizing);
+}
+
+// static
+mate::WrappableBase* ResizeArea::New(mate::Arguments* args) {
+  // Constructor call.
+  auto* view = new ResizeArea();
+  view->InitWith(args->isolate(), args->GetThis());
+  return view;
```

Comment:
I'm confused how `ResizeArea` can be cast to `mate::WrappableBase` given that it doesn't inherit from `mate::Wrappable`?

Reply:
This class inherits from `api::View`, which inherits from `mate::Wrappable`.

Reply:
oh, d'oh, I thought `View` was `views::View`. Confusing.